File: lession3/FlyWeight/java/NoPattern/NoPatternExtension/src/main/java/ColorFactory.java
import java.util.HashMap;
import java.util.Map;

/**
 * Created by Administrator on 2016/12/26.
 */
public class ColorFactory {
    private static Map<String, Color> maps = new HashMap<String, Color>();

    public static Color getColor(String key) {
        if (!maps.containsKey(key)) {
            maps.put(key, new Color(key));
        }
        return maps.get(key);
    }
}

File: lession3/FlyWeight/javascript/ColorFactory.js
var ColorFactory = (function(){
    var maps = {};
    var factory = 
    {
        getColor:function(key){
            if(!maps[key]){
                    maps[key] = new Color(key); 
            }
            return maps[key];
        }  
    };
    return factory;
})();